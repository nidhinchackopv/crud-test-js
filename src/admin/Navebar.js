import React from 'react';

const Navbar = () => {
  return (
    <nav data-testid ="nav" className="pl-10 flex justify-between text-xl pt-2 italic bg-blue-900">
      <div className=' w-20 pt-4'>
        CRUD
      </div>
      <ul className='pr-8 flex justify-between w-30 mb-2'>
            <a href ="/home" data-testid ="home" className='cursor-pointer p-2 shadow-md border-solid rounded-xl ml-8 transition duration-500 ease-in-out bg-blue-700 hover:bg-blue-300 transform hover:-translate-y-1 hover:scale-110 text-white hover:text-blue-700'>
            Home
            </a>
            <li className='cursor-pointer p-2 shadow-md border-solid rounded-xl ml-8 transition duration-500 ease-in-out bg-blue-700 hover:bg-blue-300 transform hover:-translate-y-1 hover:scale-110 text-white hover:text-blue-700'>
            Menu
            </li>
            <li className='cursor-pointer p-2 shadow-md border-solid rounded-xl ml-8 transition duration-500 ease-in-out bg-blue-700 hover:bg-blue-300 transform hover:-translate-y-1 hover:scale-110 text-white hover:text-blue-700'>
            About
            </li>
            <li className='cursor-pointer p-2 shadow-md border-solid rounded-xl ml-8 transition duration-500 ease-in-out bg-blue-700 hover:bg-blue-300 transform hover:-translate-y-1 hover:scale-110 text-white hover:text-blue-700'>
            Contact
            </li>
      </ul>
    </nav>
  );
};

export default Navbar;