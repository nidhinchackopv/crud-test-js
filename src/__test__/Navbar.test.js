import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import * as ReactDom from 'react-dom'
import Navebar from '../admin/Navebar'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';


it('can render the About page', () => {
    const tree = renderer.create(<Navebar />).toJSON();
    expect(tree).toMatchSnapshot();
})


test("contains correct text", () => {
    render(<Navebar />);

    const link = screen.getByText(/home/i);
    fireEvent.click(link);
    expect(screen.getByText("Menu"))
    expect(screen.getByText("About"))
    expect(screen.getByText("Contact"))
    expect(screen.getByTestId("nav"))
  });