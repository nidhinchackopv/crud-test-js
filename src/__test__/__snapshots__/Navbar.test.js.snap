// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render the About page 1`] = `
<nav
  className="pl-10 flex justify-between text-xl pt-2 italic bg-blue-900"
  data-testid="nav"
>
  <div
    className=" w-20 pt-4"
  >
    CRUD
  </div>
  <ul
    className="pr-8 flex justify-between w-30 mb-2"
  >
    <a
      className="cursor-pointer p-2 shadow-md border-solid rounded-xl ml-8 transition duration-500 ease-in-out bg-blue-700 hover:bg-blue-300 transform hover:-translate-y-1 hover:scale-110 text-white hover:text-blue-700"
      data-testid="home"
      href="/home"
    >
      Home
    </a>
    <li
      className="cursor-pointer p-2 shadow-md border-solid rounded-xl ml-8 transition duration-500 ease-in-out bg-blue-700 hover:bg-blue-300 transform hover:-translate-y-1 hover:scale-110 text-white hover:text-blue-700"
    >
      Menu
    </li>
    <li
      className="cursor-pointer p-2 shadow-md border-solid rounded-xl ml-8 transition duration-500 ease-in-out bg-blue-700 hover:bg-blue-300 transform hover:-translate-y-1 hover:scale-110 text-white hover:text-blue-700"
    >
      About
    </li>
    <li
      className="cursor-pointer p-2 shadow-md border-solid rounded-xl ml-8 transition duration-500 ease-in-out bg-blue-700 hover:bg-blue-300 transform hover:-translate-y-1 hover:scale-110 text-white hover:text-blue-700"
    >
      Contact
    </li>
  </ul>
</nav>
`;
